velox/functions/sparksql/DecimalArithmetic.cpp

```diff
@@ -422,12 +422,35 @@ struct DecimalMultiplyFunction {
   }
 
  private:
+  static constexpr int32_t kMaxLargeScale = 2 * LongDecimalType::kMaxPrecision;
+
+  // Pre-compute the powers of ten for large scales. The maximum scale is
+  // 2 * LongDecimalType::kMaxPrecision, which is 76. The
+  // DecimalUtil::kPowersOfTen array is not large enough to hold these values.
+  static constexpr std::array<int256_t, kMaxLargeScale + 1>
```

Comment:
How about we move this to `velox/type/DecimalUtil.h` and replace other instance of `kPowersOfTen` with `kLargeScalePowersOfTen`

Reply:
I'll take a look for this replacement.

Reply:
Replacing should suffice here

Reply:
I discovered that this was the only rescale factor that needed to be of int256 type, thus replacing it alone should be sufficient.